src/pages/index.js

```diff
@@ -34,59 +84,137 @@ profileModal.setEventListeners();
 
 profileEditButton.addEventListener("click", () => {
   const userProfileInfo = userInfo.getUserInfo();
-  profileTitleInput.value = userProfileInfo.title;
-  profileDescriptionInput.value = userProfileInfo.description;
+  profileTitleInput.value = userProfileInfo.name;
+  profileDescriptionInput.value = userProfileInfo.about;
   profileModal.open();
   formValidators["profile-form"].resetValidation();
 });
 
-function handleProfileEditSubmit({ title, description }) {
-  userInfo.setUserInfo({ title, description });
-  profileModal.close();
+function handleProfileEditSubmit(userData) {
+  profileModal.renderLoading(true);
+  api
+    .updateProfileInfo(userData.name, userData.about)
+    .then((res) => {
+      userInfo.setUserInfo(res.name, res.about);
+      profileModal.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileModal.renderLoading(false);
+    });
 }
 
-// Card Form
+// Card form
 const cardModal = new PopupWithForm("#add-card-modal", handleAddCardFormSubmit);
 cardModal.setEventListeners();
 
 addNewCardButton.addEventListener("click", () => cardModal.open());
 
-function handleAddCardFormSubmit({ name, link }) {
-  const cardElement = createCard({ name, link });
-  cardSection.addItem(cardElement);
-  addCardForm.reset();
-  cardModal.close();
-  formValidators["card-form"].disableSubmitButton();
-}
-
-// Image Preview
-const cardPreview = new PopupWithImage("#preview-image-modal");
-cardPreview.setEventListeners();
-
-function handleImageClick(cardData) {
-  cardPreview.open(cardData);
-}
-
 // Create Card
-function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
-  return card.getView();
-}
-
 const cardSection = new Section(
   {
     items: initialCards,
     renderer: (cardData) => {
-      const cardElement = createCard(cardData);
-      cardSection.addItem(cardElement);
+      cardSection.addItem(createCard(cardData));
     },
   },
   ".cards__list"
 );
 
-cardSection.renderItems();
+function createCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteButton,
+    handleLikeButton
+  );
+  return card.getView();
+}
+
+api
+  .getInitialCards()
+  .then((cards) => {
+    console.log(cards);
+    cards.forEach((card) => {
+      cardSection.addItem(createCard(card));
+    });
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+function handleAddCardFormSubmit(cardData) {
+  cardModal.renderLoading(true);
+  api
+    .addNewCard(cardData)
+    .then((res) => {
+      cardSection.addItem(createCard(res));
+      cardModal.close();
+      addCardForm.reset();
+      formValidators["card-form"].disableSubmitButton();
+    })
+    .catch((err) => {
+      console.error(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors